### PR TITLE
Adds watchdog and live updater

### PR DESCRIPTION
**Pull request by MrsTonedOne**
This is a toolset to manage a production server of tgstation13 (or its forks). It includes an update script that is able to update the server without having to stop or shutdown the server (the update will take effect next round) and a script to start the server and restart it if it crashes (optional, requires registry tweaks to disable the Windows crash dialog system wide)

This will force a live tracking of the configured git repo, so no local modifications are allowed. (this is to avoid issues with merge conflicts and because I'm too lazy to make the script detect them)

This isn't live on the server yet, and I'm not sure if this 'counts' for the feature freeze since it doesn't relate to the game code at all.

But I figured I'd add this in for commenting while I'm at work and when I get back from work I'll apply it to the server.

~~Once I make it a bit more generic and fix certain edge case issues, It would be ready for inclusion. Its configurable and repo/fork agnostic.~~

Ready

### CODE REVIEW

**Comment by Chocobro** [comment]
I think this can be exempt from feature freeze since it's system maintenance

**Comment by Miauw62** [comment]
sort of sucks that this is windows-only but oh well.

**Comment by MrStonedOne** [comment]
I made it as light on logic to make it easy as fuck to port to linux

File: tools/tgstation-server/config.bat [hunk 1]
```diff
@@ -0,0 +1,28 @@
+rem Server Tools configuration file. Lines starting with rem are comments and ignored.
+
+rem This must be set to the name of your dme without the .dme part. (should be fine leaving this alone unless you renamed the code)
+set PROJECTNAME=tgstation
+
+rem location of the repo.
+set REPO_URL=https://github.com/tgstation/-tg-station.git
+
+rem what map file to use. This should be the name of the dm, not dmm (and without the .dm part)
+set MAPFILE=metastation
```
Comment (Iamgoofball) [inline tools/tgstation-server/config.bat]: this needs to be box by default

box is our default "guarenteed to work" map

Comment (MrStonedOne) [inline tools/tgstation-server/config.bat]: Oh, forgot to change it back from when I was testing that the build script
was overwriting the ticked map file correctly

Comment (Iamgoofball) [inline tools/tgstation-server/config.bat]: sounds good

**Comment by vista-** [comment]
If I have the time on the weekend, I'll make a *nix port.

**Comment by d3athrow** [comment]
interesting implementation, dont know why you didnt just use nexis' watchdog though.

**Comment by MrStonedOne** [comment]
@d3athrow mainly because I hate python as a language, mainly because this will just work in windows 7 without having to install a language, configure it, path it, then install like 50 million dependencies, and also because I want to be able to customize it.

This installs updates LIVE(kinda), no need to stop or restart dd, users experience no interruption, at the next round, the new update will be used during a normal world restart.

**Comment by vista-** [comment]
I doubt installing Python on the server would be a challenging task. Portability > convenience.

**Comment by MrStonedOne** [comment]
To me? no To some guy who wants to "set up his own server" maybe.

Installing python isn't actually that hard, getting it in the path and installing a program's dependencies on the other hand....

Its just another item on a long list of items that anybody who looks at setting up a server sees and starts to get overwhelmed by.

Mine requires git for windows be installed (and byond of course).

After that, its click, wait, done.

**Comment by optimumtact** [comment]
Watchdog isn't a required step however, it's an optional think as an ease of life feature for admins.

Either way, as long as we have something, we barely support linux as is with no bygex linux lib

**Comment by vista-** [comment]
@optimumtact I actually have a libbygex.so compiled, if you are interested in testing it, PM me on IRC (I might be offline as it is just a bouncer).

**Comment by optimumtact** [comment]
This isn't getting merged because you besmirched the name of python

**Pull request merged**